dace/codegen/targets/cpp.py

```diff
@@ -209,22 +210,29 @@ def memlet_copy_to_absolute_strides(dispatcher, sdfg, memlet, src_node, dst_node
     return copy_shape, src_strides, dst_strides, src_expr, dst_expr
 
 
-def ptr(name: str, desc: data.Data, sdfg: SDFG = None) -> str:
+def ptr(name: str, desc: data.Data, sdfg: SDFG = None, framecode=None) -> str:
     """
     Returns a string that points to the data based on its name and descriptor.
     :param name: Data name.
     :param desc: Data descriptor.
     :return: C-compatible name that can be used to access the data.
     """
+    from dace.codegen.targets.framecode import DaCeCodeGenerator  # Avoid import loop
+    framecode: DaCeCodeGenerator = framecode
+
     # Special case: If memory is persistent and defined in this SDFG, add state
     # struct to name
-    if (desc.transient and desc.lifetime is dtypes.AllocationLifetime.Persistent
-            and desc.storage != dtypes.StorageType.CPU_ThreadLocal):
-        from dace.codegen.targets.cuda import CUDACodeGen  # Avoid import loop
-        if not CUDACodeGen._in_device_code:  # GPU kernels cannot access state
-            if not sdfg:
-                raise ValueError("Missing SDFG value")
-            return f'__state->__{sdfg.sdfg_id}_{name}'
+    if desc.storage != dtypes.StorageType.CPU_ThreadLocal:
+        if (desc.transient and desc.lifetime is dtypes.AllocationLifetime.Persistent):
+            from dace.codegen.targets.cuda import CUDACodeGen  # Avoid import loop
+            if not CUDACodeGen._in_device_code:  # GPU kernels cannot access state
+                return f'__state->__{sdfg.sdfg_id}_{name}'
```

Comment:
Is it possible to arrive here with `sdfg is None`? I guess it raises an exception anyway (before and after code change), so maybe not so important.